travis/tc3_style/check.sh

```diff
@@ -4,28 +4,40 @@ source $CI_HELPER_PATH/travis/settings.sh
 
 exit_code=0
 
-_header "Checking that source code files do not contain tabs..."
+_header "Checking that source code files do not contain leading tabs..."
 
-tab_lines=$(./files.sh | xargs awk '/\t/' | wc -l)
-if [ "${tab_lines}" -ne 0 ]; then
-  echo "Found ${tab_lines} lines with tabs"
+tab_lines=$(./files.sh | xargs egrep $'^\s*\t')
+if [ -n "${tab_lines}" ]; then
+  tab_count=$(echo "${tab_lines}" | wc -l)
+  echo "Found ${tab_count} lines with leading tabs"
+  echo "${tab_lines}"
   exit_code=1
+else
+  echo "Found no lines with tabs!"
 fi
 
 _header "Checking for lines with trailing whitespace..."
 
-bad_whitespace_lines=$(./files.sh | xargs egrep $' +$| +\r$' | wc -l)
-if [ "${bad_whitespace_lines}" -ne 0 ]; then
-  echo "Found ${bad_whitespace_lines} lines with trailing whitespace"
+whitespace_lines=$(./files.sh | xargs egrep $' +$| +\r$')
```

Comment:
Dumb question - what's the purpose of the `$'...'` at the beginning of the regex?

Reply:
The internet suggested it to get the \r to be interpreted as a carriage return. Apparently it makes the string use "c-style escapes". See https://superuser.com/questions/194668/grep-to-find-files-that-contain-m-windows-carriage-return